.github/workflows/spelling2.yml

```diff
@@ -3,20 +3,70 @@ name: Spell checking
 on:
   pull_request_target:
   push:
+    branches: ["**"]
+    tags-ignore: ["**"]
 
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+    outputs:
+      internal_state_directory: ${{ steps.spelling.outputs.internal_state_directory }}
     runs-on: ubuntu-latest
+    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
     steps:
     - name: checkout-merge
       if: "contains(github.event_name, 'pull_request')"
       uses: actions/checkout@v2
       with:
         ref: refs/pull/${{github.event.pull_request.number}}/merge
     - name: checkout
-      if: "!contains(github.event_name, 'pull_request')"
+      if: github.event_name == 'push'
       uses: actions/checkout@v2
-    - uses: check-spelling/check-spelling@v0.0.19
+    - name: check-spelling
+      id: spelling
+      uses: check-spelling/check-spelling@v0.0.20-alpha3
       with:
         config: .github/actions/spell-check
+        suppress_push_for_open_pull_request: 1
+        post_comment: 0
+    - name: store-comment
+      if: failure()
+      uses: actions/upload-artifact@v2
+      with:
+        retention-days: 1
+        name: "check-spelling-comment-${{ github.run_id }}"
+        path: |
+          ${{ steps.spelling.outputs.internal_state_directory }}
+
+  comment:
+    name: Comment
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
The write token is only used on this side, where it's no longer actively processing data, and instead only taking the generated comment data from the previous stage.

Reply:
Why does it need these permissions? It didn't seem to need write permissions before, did it?